### PR TITLE
Integrate new `test-infra` version based on components

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

This PR integrates all the modifications done in `test-infra` and change all existing tests to be based on components.
Tests that were not using `TestSuite` are barely impacted.

See the docstring in `pkg/e2e/suite.go` for details on concepts.

### Motivation

Integrate the work done in https://github.com/DataDog/test-infra-definitions/pull/484 in `datadog-agent` E2E Tests.

### Additional Notes

It also fixes several issues (StackManager accounts for `test` deadline to avoid locked stacks, etc.)

### Describe how to test/QA your changes

E2E Tests are running OK.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: test/new-e2e/tests/process/linux_test.go [hunk 1]
```diff
@@ -13,38 +13,36 @@ import (
...
 }
 
 func (s *linuxTestSuite) SetupSuite() {
-	s.Suite.SetupSuite()
-
 	// Start a process and keep it running
-	s.Env().VM.Execute("sudo apt-get -y install stress")
-	s.Env().VM.Execute("nohup stress -d 1 >myscript.log 2>&1 </dev/null &")
+	s.Env().Host.Execute("sudo apt-get -y install stress")
+	s.Env().Host.Execute("nohup stress -d 1 >myscript.log 2>&1 </dev/null &")
```
Comment (robertjli) [inline test/new-e2e/tests/process/linux_test.go:32]: Should these be `MustExecute()`?

Comment (vboulineau) [reply test/new-e2e/tests/process/linux_test.go:32]: Indeed

File: test/new-e2e/pkg/utils/e2e/client/agent_commands.go [hunk 2]
```diff
@@ -6,152 +6,157 @@
...
+	isReady  bool
 }
 
 // Create a new instance of agentCommandRunner
-func newAgentCommandRunner(t *testing.T, executeAgentCmdWithError executeAgentCmdWithError) *agentCommandRunner {
-	agent := &agentCommandRunner{
-		t:                        t,
-		executeAgentCmdWithError: executeAgentCmdWithError,
-		isReady:                  false,
+func newAgentCommandRunner(t *testing.T, executor agentCommandExecutor) *agentCommands {
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/e2e/client/agent_commands.go:34]: **💭 thought**
I preferred `agentCommandRunner`, more explicitly states that can run commands, rather than return command strings

File: test/new-e2e/pkg/utils/e2e/client/agent_docker.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+func newAgentDockerExecutor(dockerClient *Docker, agentContainerName string) *agentDockerExecutor {
+	return &agentDockerExecutor{
+		dockerClient:       dockerClient,
+		agentContainerName: agentContainerName,
+	}
+}
+
+func (ae agentDockerExecutor) execute(arguments []string) (string, error) {
+	// We consider that in a container, "agent" is always in path and is the single entrypoint.
+	// It's mostly incorrect but it's what we have ATM.
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/e2e/client/agent_docker.go:24]: **❓ question**
Why is this incorrect ?

File: test/new-e2e/pkg/utils/e2e/client/docker.go [hunk 4]
```diff
@@ -128,20 +105,8 @@ func (docker *Docker) GetClient() *client.Client {
...
-func (docker *Docker) executeAgentCmdWithError(commands []string) (string, error) {
-	wholeCommands := []string{"agent"}
-	wholeCommands = append(wholeCommands, commands...)
-	return docker.ExecuteCommandWithErr(docker.GetAgentContainerName(), wholeCommands...)
-}
+// func (docker *Docker) executeAgentCmdWithError(commands []string) (string, error) {
+// 	wholeCommands := []string{"agent"}
+// 	wholeCommands = append(wholeCommands, commands...)
+// 	return docker.ExecuteCommandWithErr(docker.GetAgentContainerName(), wholeCommands...)
+// }
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/e2e/client/docker.go:112]: **🥜 nitpick**
```suggestion
```

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 5]
```diff
@@ -37,6 +37,10 @@ const (
 )
 
 var (
+	defaultWorkspaceEnvVars = map[string]string{
+		"PULUMI_SKIP_UPDATE_CHECK": "true",
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:41]: **👏 praise**

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 6]
```diff
@@ -261,7 +258,13 @@ func buildWorkspace(ctx context.Context, profile runner.Profile, stackName strin
 		return nil, fmt.Errorf("unable to create temporary folder at: %s, err: %w", workspaceStackDir, err)
 	}
 
-	return auto.NewLocalWorkspace(ctx, auto.Project(project), auto.Program(runFunc), auto.WorkDir(workspaceStackDir))
+	fmt.Printf("Creating workspace for stack: %s at %s", stackName, workspaceStackDir)
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:261]: **💬 suggestion**
What about passing `t` context and using `t.Log` instead ?

File: test/new-e2e/pkg/utils/optional/optional.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package optional
+
+func MakeParams[Param any, Option ~func(*Param) error](options ...Option) (*Param, error) {
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/optional/optional.go:8]: **👏 praise**
TIL: https://go.dev/ref/spec#Underlying_types

File: test/new-e2e/test-infra-definition/cross_platform_test.go [hunk 8]
```diff
@@ -34,36 +36,27 @@ func (s *crossPlatformSuite) TestWindows() {
...
 		s.subTestInstallAgent(os)
 	}
 }
 
-func (s *crossPlatformSuite) subTestInstallAgent(os ec2os.Type) {
+func (s *crossPlatformSuite) subTestInstallAgent(os os.Descriptor) {
 	s.T().Run(fmt.Sprintf("Test install agent %v", os), func(t *testing.T) {
-		s.UpdateEnv(e2e.AgentStackDef(e2e.WithVMParams(ec2params.WithOS(os))))
-		output := s.Env().Agent.Status().Content
+		s.UpdateEnv(awsvm.Provisioner(awsvm.WithEC2VMOptions(ec2.WithOS(os))))
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/cross_platform_test.go:53]: **💭 thought**
We should change this test to follow guidelines of cross-platform tests

File: test/new-e2e/test-infra-definition/docker_test.go [hunk 9]
```diff
@@ -10,28 +10,28 @@ import (
...
 )
 
 type dockerSuite struct {
-	e2e.Suite[e2e.DockerEnv]
+	e2e.BaseSuite[environments.DockerVM]
 }
 
 func TestDocker(t *testing.T) {
-	e2e.Run(t, &dockerSuite{}, e2e.DockerStackDef(dockerparams.WithAgent()))
+	e2e.Run(t, &dockerSuite{}, e2e.WithProvisioner(awsdocker.Provisioner(awsdocker.WithoutFakeIntake())))
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/docker_test.go:24]: **🔨 warning**
Requires rebase / main merge

File: test/new-e2e/tests/agent-platform/install-script/install_script_test.go [hunk 10]
```diff
@@ -10,57 +10,40 @@ import (
...
-		"centos":      ec2os.CentOS,
-		"amazonlinux": ec2os.AmazonLinuxOS,
-		"redhat":      ec2os.RedHatOS,
-		"rhel":        ec2os.RedHatOS,
-		"sles":        ec2os.SuseOS,
-		"windows":     ec2os.WindowsOS,
-		"fedora":      ec2os.FedoraOS,
-		"suse":        ec2os.SuseOS,
-		"rocky":       ec2os.RockyLinux,
-	}
```
Comment (pducolin) [inline test/new-e2e/tests/agent-platform/install-script/install_script_test.go:57]: **❓ question**
Where did this go ?

cc @KevinFairise2

File: test/new-e2e/tests/agent-subcommands/diagnose_test.go [hunk 11]
```diff
@@ -34,7 +38,7 @@ var allSuites = []string{
 }
 
 func TestAgentDiagnoseEC2Suite(t *testing.T) {
-	e2e.Run(t, &agentDiagnoseSuite{}, e2e.FakeIntakeStackDef(e2e.WithFakeIntakeParams(fakeintakeparams.WithoutLoadBalancer())))
+	e2e.Run(t, &agentDiagnoseSuite{}, e2e.WithProvisioner(awsvm.Provisioner(awsvm.WithFakeIntakeOptions(fakeintake.WithoutLoadBalancer()))))
```
Comment (pducolin) [inline test/new-e2e/tests/agent-subcommands/diagnose_test.go:41]: **🔨 warning**
`WithoutLoadBalancer` is now the default

**Review by pducolin — COMMENTED** [review]
Will let test owners review their test. On the pkg side my feeling is that moving from `VM` to `Host` adds no value and adds confusion to those who are used to using `VM`

I would also duplicate envs to have one with just the VM, one with VM+agent, one with Vm + agent + fakeintake for compatibility to what people are using. 

Would limit changes to strictly required, so would keep fakeintake changes to a separate PR. 

Other than that this PR adds usability to the test framework, thank you for doing this.

File: test/fakeintake/client/client.go [hunk 12]
```diff
@@ -40,12 +40,12 @@ package client
 
 import (
 	"bytes"
+	"crypto/tls"
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply test/fakeintake/client/client.go:43]: It's because `test-infra` may now return a full https URL, so yes it is required, though the changes are in dedicated commit.

File: test/fakeintake/client/client.go [hunk 13]
```diff
@@ -267,56 +270,6 @@ func (c *Client) FilterMetrics(name string, options ...MatchOpt[*aggregator.Metr
 	return filteredMetrics, nil
 }
 
-// WithTags filters by `tags`
-func WithTags[P aggregator.PayloadItem](tags []string) MatchOpt[P] {
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply test/fakeintake/client/client.go:271]: Not removed, just moved to `filters.go` as they were in-between `Client` member functions while not directly related to `Client` code.

File: test/new-e2e/pkg/components/host_params.go [hunk 14]
```diff
@@ -19,41 +18,25 @@ import (
...
 
-// NewParams creates a new instance of Execute params
-// default Env: map[string]string{}
-func NewParams(options ...Option) (*Params, error) {
-	p := &Params{
-		EnvVariables: map[string]string{},
-	}
-	return applyOption(p, options...)
-}
+var envVarNameRegexp = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*$")
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply test/new-e2e/pkg/components/host_params.go:24]: I think it's what I did already, previously it was re-created all the time.

File: test/new-e2e/pkg/utils/infra/stack_manager.go (hunk 6 above)
Comment (vboulineau) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:261]: I don't think that `stackManager` should be test-aware, but with the changes done to pass a logger, we can implement that.

File: test/new-e2e/tests/agent-platform/install-script/install_script_test.go (hunk 10 above)
Comment (vboulineau) [reply test/new-e2e/tests/agent-platform/install-script/install_script_test.go:57]: We do this defined in test-infra already: https://github.com/DataDog/test-infra-definitions/blob/6295abafbaa844758b21a9e0e6a946aad1d16846/components/os/const.go

File: test/fakeintake/client/client.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply test/fakeintake/client/client.go:43]: Are the retry changes required too ?

File: test/new-e2e/examples/agentenv_copyfolder_test.go [hunk 15]
```diff
@@ -10,39 +10,37 @@ import (
...
 
 type agentSuiteEx6 struct {
-	e2e.Suite[e2e.AgentEnv]
+	e2e.Suite[environments.VM]
 }
 
 func TestAgentSuiteEx6(t *testing.T) {
-
-	e2e.Run(t, &agentSuiteEx6{}, e2e.AgentStackDef(e2e.WithVMParams(ec2params.WithOS(ec2os.UbuntuOS))))
+	e2e.Run(t, &agentSuiteEx6{}, e2e.WithProvisioner(awsvm.Provisioner()))
```
[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/examples/agentenv_copyfolder_test.go:25]: I see your point, we used this as an example to show how to use the vm params. We have another example specific, we could keep it there, but bringing it back there.

File: test/fakeintake/client/client.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply test/fakeintake/client/client.go:43]: Could we change the implementation on `test-infra` side to avoid breaking changes on this side ?

File: test/new-e2e/pkg/utils/e2e/client/agent_docker.go (hunk 3 above)
Comment (vboulineau) [reply test/new-e2e/pkg/utils/e2e/client/agent_docker.go:24]: Because you may want to run commands for any agent binary and the different agent binaries are in the same folder.

File: test/new-e2e/.vscode/tasks.json.template [hunk 16]
```diff
@@ -5,7 +5,7 @@
         {
             "label": "aws-vault-wrap",
             "type": "shell",
-            "command": "aws-vault export sso-agent-sandbox-account-admin > .vscode/aws-vault-wrap.env",
+            "command": "aws-vault export sso-agent-sandbox-account-admin > ${workspaceFolder}/.vscode/aws-vault-wrap.env",
```
Comment (pducolin) [inline test/new-e2e/.vscode/tasks.json.template:8]: **❓ question**
Does this wrap terminal only, or also go.tools ?

File: test/new-e2e/examples/agentenv_logs_test.go [hunk 17]
```diff
@@ -7,41 +7,49 @@ package examples
...
 		logs, err := fakeintake.FilterLogs("custom_logs")
 		assert.NoError(c, err)
 		assert.Equal(c, len(logs), 0, "logs received while none expected")
 	}, 5*time.Minute, 10*time.Second)
-	// part 2: generate logs
-	s.Env().VM.Execute("echo 'totoro' > /tmp/test.log")
-	// part 3: there should be logs
 	s.EventuallyWithT(func(c *assert.CollectT) {
+		// part 2: generate logs
+		s.Env().RemoteHost.MustExecute("echo 'totoro' >> /tmp/test.log")
```
Comment (pducolin) [inline test/new-e2e/examples/agentenv_logs_test.go:51]: **❓ question**
Why is this moved inside the loop ? It is going to generate n logs, while we want to generate once and wait for the log to reach the fakeintake

File: test/new-e2e/pkg/e2e/suite_test.go [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+	s.tempProvisioner.AssertNumberOfCalls(s.T(), "Provision", 2)
+	s.tempProvisioner.AssertNumberOfCalls(s.T(), "Destroy", 1)
+}
+
+func (s *testSuiteWithTests) TestOrderC() {
+	// The provisioners were not changed, so nothing should happen
+	s.permanentProvisioner.AssertNumberOfCalls(s.T(), "Provision", 3)
+	s.tempProvisioner.AssertNumberOfCalls(s.T(), "Provision", 2)
+	s.tempProvisioner.AssertNumberOfCalls(s.T(), "Destroy", 1)
+}
```
Comment (pducolin) [inline test/new-e2e/pkg/e2e/suite_test.go:157]: **👏 praise**

File: test/new-e2e/pkg/environments/aws/host/host.go [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+// WithoutAgent disables the creation of the Agent.
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// ProvisionerNoAgentNoFakeIntake wraps Provisioner with hardcoded WithoutAgent and WithoutFakeIntake options.
+func ProvisionerNoAgentNoFakeIntake(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.Host] {
```
Comment (pducolin) [inline test/new-e2e/pkg/environments/aws/host/host.go:109]: **👏 praise**
Thank you !

File: test/new-e2e/test-infra-definition/cross_platform_test.go [hunk 20]
```diff
@@ -9,23 +9,25 @@ import (
...
 }
 
 func TestCrossPlatformSuite(t *testing.T) {
-	e2e.Run(t, &crossPlatformSuite{}, e2e.AgentStackDef())
+	e2e.Run(t, &crossPlatformSuite{}, e2e.WithProvisioner(awshost.Provisioner(awshost.WithoutFakeIntake())))
 }
 
 func (s *crossPlatformSuite) TestUbuntuOS() {
-	os := ec2os.UbuntuOS
+	os := os.UbuntuDefault
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/cross_platform_test.go:30]: **💭 thought**
Note for myself: refactor this test to use one suite per os

**Review by pducolin — APPROVED** [review]
Thank you for addressing the review, there is one question with a possible introduced bug in on example test (logs)

File: test/new-e2e/.vscode/tasks.json.template (hunk 16 above)
Comment (vboulineau) [reply test/new-e2e/.vscode/tasks.json.template:8]: The config is not new, it was just not fully working. It's only used for `debug` run.

File: test/new-e2e/examples/agentenv_logs_test.go (hunk 17 above)
Comment (vboulineau) [reply test/new-e2e/examples/agentenv_logs_test.go:51]: Because the test was flaky, I did not fully investigate but as it is an example, I prefer to have it working always.

**Review by paulcacheux — APPROVED** [review]

**Review by carlosroman — APPROVED** [review]

File: test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go [hunk 21]
```diff
@@ -9,24 +9,24 @@ package flare
...
 )
 
-type windowsFlareSuite struct {
-	baseFlareSuite
-}
+type windowsFlareSuite baseFlareSuite
 
 func TestWindowsFlareSuite(t *testing.T) {
-	e2e.Run(t, &windowsFlareSuite{}, e2e.FakeIntakeStackDef(e2e.WithVMParams(ec2params.WithOS(ec2os.WindowsOS))))
+	t.Parallel()
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go:24]: I think we already run tests in different jobs, I would rather avoid this unexpected change

File: test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go [hunk 22]
```diff
@@ -9,24 +9,24 @@ package flare
...
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+
 	"github.com/stretchr/testify/assert"
 )
 
-type windowsFlareSuite struct {
-	baseFlareSuite
-}
+type windowsFlareSuite baseFlareSuite
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go:21]: I'm not sure this is strictly equivalent (eg. for casting one type to another), any reason for this change ?

File: test/new-e2e/tests/agent-subcommands/flare/flare_nix_test.go [hunk 23]
```diff
@@ -27,24 +26,22 @@ var systemProbeConfiguration []byte
...
 var securityAgentConfiguration []byte
 
-type linuxFlareSuite struct {
-	baseFlareSuite
-}
+type linuxFlareSuite baseFlareSuite
 
 func TestLinuxFlareSuite(t *testing.T) {
-	e2e.Run(t, &linuxFlareSuite{}, e2e.FakeIntakeStackDef(e2e.WithVMParams(ec2params.WithOS(ec2os.UbuntuOS))))
+	t.Parallel()
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-subcommands/flare/flare_nix_test.go:32]: I think we already run tests in different jobs, I would rather avoid this unexpected change

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go [hunk 24]
```diff
@@ -18,33 +18,47 @@ import (
...
+	return nil
 }
 
 const (
 	// intakeName should only contains alphanumerical characters to ease pattern matching /etc/hosts
 	intakeName      = "ddintake"
-	fakeintake1Name = "1fakeintake"
-	fakeintake2Name = "2fakeintake"
+	fakeintake1Name = "1"
+	fakeintake2Name = "2"
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:61]: Any particular reason for the renaming ? Having a distinctive name was very convenient when reading the logs of the test...

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go [hunk 25]
```diff
@@ -68,48 +82,49 @@ var customLogsConfigTmplFile string
...
+func multiFakeIntakeAWS(agentOptions ...agentparams.Option) e2e.Provisioner {
+	runFunc := func(ctx *pulumi.Context, env *multiFakeIntakeEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		host, err := ec2.NewVM(awsEnv, "nssfailover")
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:92]: The VM doesn't use a default name like it used to ?

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go [hunk 26]
```diff
@@ -68,48 +82,49 @@ var customLogsConfigTmplFile string
...
 		if err != nil {
-			return nil, err
+			return err
 		}
+		host.Export(ctx, &env.Host.HostOutput)
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:96]: **🥜 nitpick:** ‏Seems weird that creating any resource is followed by calling the Export function, I would expect for example a wrapper function to handle that transparently if it is really needed

(I know creating everything like we do in this test is uncommon so it's not a big deal honestly)

Comment (vboulineau) [reply test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:96]: It's not creating any resource, it's creating resources that need to be exported, hence the `Export`. In that case all resources need to be exported. It's not always the case.

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go (hunk 25 above)
Comment (vboulineau) [reply test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:92]: Indeed, `name` is a mandatory parameter for any resource, there's no reason not to expose it as such.
A name is provided in pre-built scenarios.

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go (hunk 24 above)
Comment (vboulineau) [reply test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:61]: The component now adds the `fakeintake` [part itself](https://github.com/DataDog/test-infra-definitions/blob/b181ca52249a54e12931386f368539883e435ec2/scenarios/aws/fakeintake/fakeintake.go#L46-L47), so removed here to avoid duplication;

File: test/new-e2e/tests/agent-subcommands/flare/flare_nix_test.go (hunk 23 above)
Comment (vboulineau) [reply test/new-e2e/tests/agent-subcommands/flare/flare_nix_test.go:32]: Merge conflict, thanks you for spotting.
`t.Parallel()` used to be there: https://github.com/DataDog/datadog-agent/commit/b6901b6bc5e403035b9b411dc9508d992a0c4d9c#diff-752a805b7748973073858f2aec7ffbe8dc468162f62357acbe485d9a8b5ef860L36

File: test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go (hunk 26 above)
Comment (pgimalac) [reply test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go:96]: Hum ok

File: test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go (hunk 22 above)
Comment (vboulineau) [reply test/new-e2e/tests/agent-subcommands/flare/flare_win_test.go:21]: It's not equivalent indeed (not only in terms of cast. Both works in this case, but still not the same).
We should favour embed, this is a remain from some testing with type declaration.

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for files owned by @DataDog/container-integrations.

**Review by richl-dd — APPROVED** [review]

**Review by jmw51798 — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

**Review by clarkb7 — APPROVED** [review]

File: test/new-e2e/tests/process/linux_test.go [hunk 27]
```diff
@@ -9,43 +9,43 @@ import (
...
-	e2e.Suite[e2e.FakeIntakeEnv]
+	e2e.BaseSuite[environments.Host]
 }
 
 func TestLinuxTestSuite(t *testing.T) {
-	e2e.Run(t, &linuxTestSuite{},
-		e2e.FakeIntakeStackDef(
-			e2e.WithAgentParams(agentparams.WithAgentConfig(processCheckConfigStr)),
-		))
+	e2e.Run(t, &linuxTestSuite{}, e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(agentparams.WithAgentConfig(processCheckConfigStr)))), e2e.WithDevMode())
```
Comment (robertjli) [inline test/new-e2e/tests/process/linux_test.go:27]: Was DevMode intentionally added here?

**Review by robertjli — APPROVED** [review]
Left one question, otherwise LGTM for processes 🙇

File: test/new-e2e/tests/process/linux_test.go [hunk 28]
```diff
@@ -9,43 +9,43 @@ import (
...
 }
 
 func (s *linuxTestSuite) SetupSuite() {
-	s.Suite.SetupSuite()
+	s.BaseSuite.SetupSuite()
 
 	// Start a process and keep it running
-	s.Env().VM.Execute("sudo apt-get -y install stress")
-	s.Env().VM.Execute("nohup stress -d 1 >myscript.log 2>&1 </dev/null &")
+	s.Env().RemoteHost.MustExecute("sudo apt-get -y install stress")
```
Comment (pducolin) [inline test/new-e2e/tests/process/linux_test.go:34]: @robertjli @vboulineau this should go to the run function, as part of setup
